### ISSUE TITLE
Ticking "show config" in a preset

### ISSUE THREAD

**Issue by quantumm** [PyMoDAQ/PyMoDAQ#350]
Hi everyone,

I found a strange bug while defining a preset with a 1D detector. In the Main settings, if I tick Show config and save the preset (don’t ask why I did that!)

![Capture_show_config_bug](https://github.com/PyMoDAQ/PyMoDAQ/assets/5221986/4db9a13e-4adb-48ec-8421-db71f2948762)

then when I want to load the preset, it will freeze at the pymodaq pop-up window. It looks like it wants to open the template configuration popup window at this moment...

A quick answer would be to not tick Show config while defining a preset, which is useless, but i guess there should be a clever way to fix that ;)

**Comment by seb5g** [comment]
this is more or less related also to the interplay between config files and preset config. Should we define all these numbers in the preset or in a plugin config file. Then at the moment of initialization, the developper could read those config value and apply them in the right order. Then we could only have the preset to config the master/slave stuff. 

We already discuss that a bit at the pymodaq days with @rgeneaux  @Ashwolaa et @aurore-finco @jerlfan  who were there. What are your thought on that?

**Comment by seb5g** [comment]
@rgeneaux

**Comment by Ashwolaa** [comment]
> this is more or less related also to the interplay between config files and preset config. Should we define all these numbers in the preset or in a plugin config file. Then at the moment of initialization, the developper could read those config value and apply them in the right order. Then we could only have the preset to config the master/slave stuff. 
> 
> We already discuss that a bit at the pymodaq days with @rgeneaux  @Ashwolaa et @aurore-finco @jerlfan  who were there. What are your thought on that?  

I think that the numbers should be in the plugin configuration files, the preset should only allow you to look for a configuration or to make one.

So I would just imagine two buttons in the preset, one to select a configuration file from a plugin or one to make one

**Comment by quantumm** [comment]
I think I agree with that.

**Comment by jerlfan** [comment]
I am agree with that too.

**Comment by aurore-finco** [comment]
I agree that I would prefer having as less things as possible in the preset, because I had several times the issue that my presets were not compatible anymore after updating pymodaq, and creating them again was quite long, inputing all the numbers, finding again which channels to use on the NI card, etc.
Keeping the preset only to select instruments with their config and set up the master/slave connections seems good to me.

**Comment by seb5g** [comment]
All these trouble will disapear soon: the  new preset and the configurator are out in the dev branch. You can use that, test it and tell if the usage is better!! But I bet it will be

**Comment by seb5g** [comment]
no more relevant for >=5.2
